sale_coupon_multi_use/models/sale_coupon.py

```diff
@@ -40,6 +49,14 @@ def _compute_discount_fixed_amount_delta(self):
         for rec in self:
             rec.discount_fixed_amount_delta = rec._get_discount_fixed_amount_delta()
 
+    def _check_coupon_code(self, order):
+        # Same check as is for `applied_coupon_ids` field.
+        if self.program_id in order.coupon_multi_use_ids.mapped("program_id"):
```

Comment:
You don't need to load `coupon_multi_use_ids` to use the mapped function.
Let Odoo manage the mapped entirely.
```suggestion
        if self.program_id in order.mapped("coupon_multi_use_ids.program_id"):
```

Reply:
Is there a difference?

Reply:
Yes, you let Odoo manage data loading.
In your case, you load all `coupon_multi_use_ids` to access to programs.
In the full mapped, you delegate this to Odoo, and maybe it's better optimized, maybe not, but it's the responsibility of Odoo, and if in a future patch they optimize or changed something in the mapped implementation, you will benefit of it.

Reply:
Hm, but this was copy/pasted from this `self.program_id in order.applied_coupon_ids.mapped('program_id'):` (original check) Just changed field used, but case is basically the same.

Reply:
The Odoo code is not always a good reference.